libraries/alicloud_backend.rb

```diff
@@ -185,11 +186,14 @@ def validate_parameters(allow: [], required: nil, require_any_of: nil)
       allow += require_any_of
     end
 
-    allow += %i{region endpoint}
+    allow += %i{region} unless allow.include?(:region)
+    allow += %i{endpoint} unless allow.include?(:endpoint)
+    @opts.delete(:region) if @opts.is_a?(Hash) && @opts[:region].nil?
+
     raise ArgumentError, "Scalar arguments not supported" unless defined?(@opts.keys)
     raise ArgumentError, "Unexpected arguments found" unless @opts.keys.all? { |a| allow.include?(a) }
     raise ArgumentError, "Provided parameter should not be empty" unless @opts.values.all? do |a|
-      return true if a.class == Integer
+      return true if a.instance_of?(Integer) || a.instance_of?(TrueClass) || a.instance_of?(FalseClass)
```

Comment:
```suggestion
      return true if (a.respond_to?(:empty?) ? !a.empty? : !a)
```

Reply:
you can remove this [line](https://github.com/inspec/inspec-alicloud/pull/53/files#diff-533ce34f00798db00b4356b57c535ac15a505d19d572cf4be6703594c43145f2R198) now